### ISSUE TITLE
Add ArcFace loss.

### ISSUE THREAD

**Issue by innat** [keras-team/keras#16701]
**System information**.

TensorFlow version (you are using): 2.8
Are you willing to contribute it (Yes/No) : No (but willing to provide starter if required.)

**Describe the feature and the current behavior/state**.

**ArcFace**, or Additive Angular Margin Loss, is a loss function used mainly in face recognition tasks. It is used to obtain highly discriminative features. This loss function can be used in both vision task and nlp task. So, I think its general for `keras_cv` and `keras_nlp`.

Paper: https://arxiv.org/pdf/1801.07698.pdf
Citation: 2939  (current)

**Will this change the current api? How?**

```python

tf.keras.losses.SparseArcFace
tf.keras.losses.CategoricalArcFace
```

**Who will benefit from this feature?**

`keras` practitioners. 

**[Contributing](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md)**
- Briefly describe your candidate solution(if contributing):
     - https://github.com/4uiiurz1/keras-arcface

**Comment by fchollet** [comment]
> ArcFace, or Additive Angular Margin Loss, is a loss function used mainly in face recognition tasks

What are some NLP use cases? It seems like it might be a good fit for KerasCV specifically.

**Comment by innat** [comment]
Sorry I miswrote, it should be only CV task. 
Could you please move this issue to KerasCV?
